pkg/project/handlers.go

```diff
@@ -46,21 +48,35 @@ func CreateNamespace(req router.Request, resp router.Response) error {
 
 // EnsureAllAppsRemoved ensures that all apps are removed from the project before the namespace is deleted.
 func EnsureAllAppsRemoved(req router.Request, resp router.Response) error {
-	apps := new(v1.AppInstanceList)
-	if err := req.List(apps, &kclient.ListOptions{Namespace: req.Object.GetName()}); err != nil {
+	// A "child" app is one that has a parent label. Therefore, all "parent" apps won't have a parent label.
+	parentAppRequirement, err := klabels.NewRequirement(labels.AcornParentAcornName, selection.DoesNotExist, nil)
+	if err != nil {
 		return err
 	}
 
-	for _, app := range apps.Items {
-		if app.DeletionTimestamp.IsZero() {
-			if err := req.Client.Delete(req.Ctx, &app); err != nil && !apierrors.IsNotFound(err) {
-				return err
+	// First delete all "parent" apps. If no "parent" apps exist, then ensure that all apps are deleted.
+	for _, ls := range []klabels.Selector{klabels.NewSelector().Add(*parentAppRequirement), klabels.Everything()} {
```

Comment:
`klabels.Everything()` will definitely work, but is there a specific reason this was chosen over something like `childAppRequirement`?

Reply:
Two reasons I did it this way:
1. It's obvious that we want to clean up everything; and
2. One fewer error to handle.